File: leetcode/rust-leetcode/src/1945.sum-of-digits-of-string-after-convert.rs
impl Solution {
    pub fn get_lucky(s: String, k: i32) -> i32 {
        let digit_sum = |mut n: i32| -> i32 {
            let mut sum = 0;
            while n > 0 {
                sum += n % 10;
                n /= 10;
            }
            sum
        };
        let mut n = s.into_bytes().iter().fold(0, |acc, &b| acc + digit_sum(b as i32 - 96));
        for _ in 1..k {
            n = digit_sum(n);
        }
        n
    }
}

File: leetcode/cpp/1945.sum-of-digits-of-string-after-convert.cpp
class Solution {
public:
    int getLucky(string s, int k) {
        int num = 0;
        for (char c: s) {
            num += digitSum(c - (int)'a' + 1);
        }
        while (k > 1 && num > 9) {
            num = digitSum(num);
            k--;
        }
        return num;
    }
private:
    int digitSum(int x) {
        if (x < 10) {
            return x;
        } else {
            int y = 0;
            while (x > 0) {
                y += x % 10;
                x /= 10;
            }
            return y;
        }
    }
};

